stan/math/rev/core/vari.hpp

```diff
@@ -78,27 +78,18 @@ class vari_base {
  *
  */
 template <typename T>
-class vari_value<T, require_floating_point_t<T>> : public vari_base {
+class vari_value<T, require_t<std::is_floating_point<T>>> : public vari_base {
  public:
   using value_type = std::decay_t<T>;
   /**
    * The value of this variable.
    */
-  const value_type val_;
+  const value_type val_{0.0};
```

Comment:
I think all constructors set the val, so we do not need the default. Also I don't think it is very meaningful.